.github/workflows/build_android.yml

```diff
@@ -0,0 +1,79 @@
+---
+on:
+  workflow_dispatch:
+  push:
+    paths-ignore:
+      - 'build/**'
+      - 'doc/**'
+      - 'python/**'
+      - 'subprojects/**'
+      - 'systemd/**'
+      - 'win32/**'
+    branches:
+      - master
+      - v0.23.x
+  pull_request:
+    paths-ignore:
+      - 'build/**'
+      - 'doc/**'
+      - 'python/**'
+      - 'subprojects/**'
+      - 'systemd/**'
+      - 'win32/**'
+    branches:
+      - master
+      - v0.23.x
+
+permissions:
+  contents: read #  to fetch code (actions/checkout)
+
+jobs:
+  build-android:
+    runs-on: ubuntu-22.04
+    steps:
+      - id: checkout
+        uses: actions/checkout@v4
+
+      - id: cache-ccache
+        uses: hendrikmuhs/ccache-action@v1
+        with:
+          key: android
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'zulu'
+          java-version: 17
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y --no-install-recommends \
+            ninja-build \
+            quilt 
+          pip3 install --user meson==0.56.2
```

Comment:
Meson needed a bump here to work around some include path stuff that llvm was throwing.

https://reviews.llvm.org/D131441